model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -76,79 +105,107 @@ def pattern(self):
+        new_shape = reshape_node.in_port(1).data.get_value()
+        if new_shape is None or input_rank_of_unsqueeze != len(new_shape):
+            return False
 
-        strided_slice_node.out_port(0).connect(cast_shape_to_float.in_port(0))
-        cast_shape_to_float.out_port(0).connect(mul_node.in_port(0))
+        return True
 
-        interp_node = Interpolate(graph,
-                                  dict(mode='nearest',
-                                       antialias=0, pads_begin=int64_array([0]),
-                                       pads_end=int64_array([0]), coordinate_transformation_mode='half_pixel',
-                                       nearest_mode='round_prefer_floor', cube_coeff=-0.75,
-                                       version='opset4', shape_calculation_mode='scales',
-                                       in_ports_count=4,
-                                       maybe_part_of_sequence=True)).create_node()
+    def replace_pattern(self, graph: Graph, match: dict):
+        if not self.is_applicable(match):
+            return
 
-        floor_node = Floor(graph, {'name': unsqueeze_name + '/Floor'}).create_node()
-        cast_mul_result_to_int = Cast(graph, {'dst_type': np.int64}).create_node()
+        unsqueeze_node = match['unsqueeze']
+        unsqueeze_name = unsqueeze_node.soft_get('name', unsqueeze_node.id)
+        second_input_of_unsqueeze = unsqueeze_node.in_port(1).get_connection().get_source().node
+        d_idx = int(second_input_of_unsqueeze.value)
+        axis = d_idx - 1
 
-        mul_node.out_port(0).connect(floor_node.in_port(0))
-        floor_node.out_port(0).connect(cast_mul_result_to_int.in_port(0))
+        shape_node = Shape(graph, dict(name=unsqueeze_name + '/Shape')).create_node()
+        strided_slice_node = create_op_with_const_inputs(graph,
+                                                         StridedSlice,
+                                                         {
+                                                             1: int64_array([axis]),
+                                                             2: int64_array([axis + 1])
+                                                         },
+                                                         {
+                                                             'name': unsqueeze_name + '/StridedSlice',
+                                                             'begin_mask': int64_array([1]),
+                                                             'end_mask': int64_array([1]),
+                                                             'new_axis_mask': int64_array([0]),
+                                                             'shrink_axis_mask': int64_array([0]),
+                                                             'ellipsis_mask': int64_array([0]),
+                                                         })
+        shape_node.out_port(0).connect(strided_slice_node.in_port(0))
```

Comment:
You can use function `node_to_get_shape_value_of_indices(shape_node, [axis])` from `model-optimizer/mo/utils/shape.py` instead of writing your own SS.

Reply:
Done.